### PR TITLE
Use QueryArgs for /costs route

### PR DESCRIPTION
**Pull request by giubil**
Added `QueryArgStringSlice` and `QueryArgDate` in the `routes` package, and changed the `/cost` route to use those

### CODE REVIEW

File: costs/costs_route.go [hunk 1]
```diff
@@ -56,128 +52,75 @@ type esQueryParams struct {
...
-	}
-	parsedParams, err := parseQueryParams(request.Form)
-	if err != nil {
-		l.Info("Error parsing user submitted forms : "+err.Error(), nil)
-		return esQueryParams{}, err
+// validateAwsAccounts will validate awsAccounts passed to it.
+// It checks that they are numbers that are 12 character long
+func validateAwsAccounts(parsedParams esQueryParams) error {
+	for _, account := range parsedParams.accountList {
+		if _, err := strconv.ParseInt(account, 10, 0); err != nil || len(account) != 12 {
```
Comment (schubev) [inline costs/costs_route.go:119]: Can you please point me to the documentation which says an AWS account ID is always 12 digits long ?

File: costs/costs_route.go [hunk 2]
```diff
@@ -56,128 +52,75 @@ type esQueryParams struct {
...
+			return fmt.Errorf("Error invalid account format : %s", account)
```
Comment (schubev) [inline costs/costs_route.go]: Please use the Go convention for errors.

**Review by schubev — CHANGES_REQUESTED** [review]

File: costs/costs_route.go (hunk 1 above)
Comment (giubil) [reply costs/costs_route.go:119]: It's written here : https://docs.aws.amazon.com/general/latest/gr/acct-identifiers.html

Comment (schubev) [reply costs/costs_route.go:119]: Thanks!

**Review by schubev — APPROVED** [review]

**Pull request merged**